mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,507 @@
+
+    Returns
+    -------
+    tuple, len (2)
+        Internal and external basis sets ndarrays with shape
+        (n_coils, n_mult_moments)
+    """
+    r_int_pts, ncoils, wcoils, int_pts = _concatenate_coils(coils)
+    n_sens = len(int_pts)
+    n_bases = get_num_moments(int_order, ext_order)
+    int_lens = np.insert(np.cumsum(int_pts), obj=0, values=0)
+
+    S_in = np.empty((n_sens, (int_order + 1) ** 2 - 1))
+    S_out = np.empty((n_sens, (ext_order + 1) ** 2 - 1))
+    S_in.fill(np.nan)
+    S_out.fill(np.nan)
+
+    # Set all magnetometers (with 'coil_type' == 1.0) to be scaled by 100
+    coil_scale = np.ones((len(coils)))
+    coil_scale[np.array([coil['coil_class'] == 1.0 for coil in coils])] = 100.
+
+    assert n_bases <= n_sens, ('Number of requested bases (%s) exceeds number '
+                               'of sensors (%s)' % (str(n_bases), str(n_sens)))
+
+    # Compute position vector between origin and coil integration pts
+    cvec_cart = r_int_pts - origin[np.newaxis, :]
+    # Convert points to spherical coordinates
+    cvec_sph = _cart_to_sph(cvec_cart)
+
+    # Compute internal basis vectors (exclude degree 0; RHS Eq. 5)
+    for deg in range(1, int_order + 1):
+        for order in range(-deg, deg + 1):
+
+            # Compute gradient for integration point position vectors
+            grads = -1 * _grad_in_components(deg, order, cvec_sph[:, 0],
+                                             cvec_sph[:, 1], cvec_sph[:, 2])
+
+            # Gradients dotted with integration point normals and weighted
+            a1_all = wcoils * np.einsum('ij,ij->i', grads, ncoils)
+
+            # For order and degree, sum across integration pts for each sensor
+            for pt_i in range(0, len(int_lens) - 1):
+                int_pts_sum = np.sum(a1_all[int_lens[pt_i]:int_lens[pt_i + 1]])
+                S_in[pt_i, deg ** 2 + deg + order - 1] = int_pts_sum
+
+    # Compute external basis vectors (exclude degree 0; RHS Eq. 5)
+    for deg in range(1, ext_order + 1):
+        for order in range(-deg, deg + 1):
+
+            # Compute gradient for integration point position vectors
+            grads = -1 * _grad_out_components(deg, order, cvec_sph[:, 0],
+                                              cvec_sph[:, 1], cvec_sph[:, 2])
+
+            # Gradients dotted with integration point normals and weighted
+            b1_all = wcoils * np.einsum('ij,ij->i', grads, ncoils)
+
+            # For order and degree, sum across integration pts for each sensor
+            for pt_i in range(0, len(int_lens) - 1):
+                int_pts_sum = np.sum(b1_all[int_lens[pt_i]:int_lens[pt_i + 1]])
+                S_out[pt_i, deg ** 2 + deg + order - 1] = int_pts_sum
+
+    # Scale and normalize each basis vector to have unity magnitude
+    S_in *= coil_scale[:, np.newaxis]
+    S_out *= coil_scale[:, np.newaxis]
+    S_in = np.divide(S_in, np.linalg.norm(S_in, axis=0))
+    S_out = np.divide(S_out, np.linalg.norm(S_out, axis=0))
```

Comment:
and then these can be in your loop, too. Instead of using `np.divide` just do:
```
    out *= coil_scale[:, np.newaxis]
    out /= np.sqrt(np.sum(out * out, axis=0))[np.newaxis, :]
```
Note that the `axis` argument was added in numpy 1.9, so we can't rely on it existing. Also, the `numpy` folks claim that it's faster than doing `sqrt(sum(x * x, axis=axis))` but in my BEM PR I found that it wasn't (!). So we might as well use the longer formulation.